File: instrumentation/src/main/java/io/opentelemetry/android/export/ModifiedSpanData.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.export;

import io.opentelemetry.api.common.Attributes;
import io.opentelemetry.sdk.trace.data.DelegatingSpanData;
import io.opentelemetry.sdk.trace.data.SpanData;

final class ModifiedSpanData extends DelegatingSpanData {

    private final Attributes modifiedAttributes;

    ModifiedSpanData(SpanData original, Attributes modifiedAttributes) {
        super(original);
        this.modifiedAttributes = modifiedAttributes;
    }

    @Override
    public Attributes getAttributes() {
        return modifiedAttributes;
    }

    @Override
    public int getTotalAttributeCount() {
        return modifiedAttributes.size();
    }
}

File: core/src/main/java/io/opentelemetry/android/export/ModifiedSpanData.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.export

import io.opentelemetry.api.common.Attributes
import io.opentelemetry.sdk.trace.data.DelegatingSpanData
import io.opentelemetry.sdk.trace.data.SpanData

internal class ModifiedSpanData(
    original: SpanData,
    private val modifiedAttributes: Attributes,
) : DelegatingSpanData(original) {
    override fun getAttributes(): Attributes = modifiedAttributes

    override fun getTotalAttributeCount(): Int = modifiedAttributes.size()
}
